### PR TITLE
Feature/57 update frontend to show joined groups on side bar

### PR DESCRIPTION
**Pull request by andreaestrada**
Review after feature 48, since I branched from that.

Added side panel for joined groups, cleaned up other side panel buttons to reflect ability to join groups (e.g. add create group button).

groups

### CODE REVIEW

**Comment by andreaestrada** [comment]
@zaclittleberry as with feature 48, I sometimes get the ERROR TypeError: Cannot read property 'id' of undefined at SafeSubscriber._next (app.component.ts:75). Also not sure why this happens sometimes, but seems to mostly run smoothly other than that.

File: src/app/app.component.ts
Comment (zaclittleberry) [inline src/app/app.component.ts]: This should be its own component in the /src/app/_components folder. Also, it feels a bit weird to have its logic in the app.component. I'm thinking it might be better to just make a blank page for it, have the button go to the page, and then have the ngOnInit of that page auto-open the dialog. Sorry it's kind of convoluted. We don't have any example of dialogs on core, and I've only ever triggered them from a page giving them context (not a sidebar/the main app component)

**Review by zaclittleberry — CHANGES_REQUESTED** [review]
This looks really good. I have some requests on the code organization, and then we'll need to wait until feature/48 passes to merge this as it looks like it includes those changes as well. If you're getting that error again, I'm guessing it's something similar to the change I proposed in feature/48. You probably have a race condition with getting the user account and then using that information. Restructuring in a similar way so the subscription callback for the user account triggers the dependent functions should work.

File: src/app/_pages/join/join.page.ts
Comment (zaclittleberry) [inline src/app/_pages/join/join.page.ts]: commented in slack. this should be removed and switched to some sort of event or observable to trigger the reload of data in the sidebar.

File: src/app/app.module.ts
Comment (zaclittleberry) [inline src/app/app.module.ts]: does CreateGroupDialog need to be declared here for it to work? I don't think it _should_.

Comment (andreaestrada) [reply src/app/app.module.ts]: I got the error: "Component CreateGroupDialog is not part of any NgModule or the module has not been imported into your module" when I removed CreateGroupDialog. Not sure why it is required though, maybe I have imports wrong somewhere else?

File: src/app/_pages/join/join.page.ts
Comment (andreaestrada) [reply src/app/_pages/join/join.page.ts]: Fixed!

**Review by zaclittleberry — DISMISSED** [review]

**Review by zaclittleberry — APPROVED** [review]

**Pull request merged**
